lib/components/user/account-page.js

```diff
@@ -1,26 +1,67 @@
-import React from 'react'
+import { withAuthenticationRequired } from '@auth0/auth0-react'
+import React, { Component } from 'react'
 import { Col, Row } from 'react-bootstrap'
+import { connect } from 'react-redux'
 
+import * as uiActions from '../../actions/ui'
+import { CREATE_ACCOUNT_PATH } from '../../util/constants'
+import { RETURN_TO_CURRENT_ROUTE } from '../../util/ui'
+import withLoggedInUserSupport from './with-logged-in-user-support'
 import DesktopNav from '../app/desktop-nav'
 import SubNav from './sub-nav'
 
 /**
  * This component contains common navigation elements and wrappers and should
  * wrap any user account page (e.g., SavedTripList or account settings).
  */
-const AccountPage = ({children, subnav = true}) => (
-  <div className='otp'>
-    {/* TODO: Do mobile view. */}
-    <DesktopNav />
-    {subnav && <SubNav />}
-    <div className='container'>
-      <Row xs={12}>
-        <Col xs={12} sm={10} smOffset={1} md={8} mdOffset={2}>
-          {children}
-        </Col>
-      </Row>
-    </div>
-  </div>
-)
+class AccountPage extends Component {
+  componentDidMount () {
+    const { loggedInUser, routeTo } = this.props
+
+    if (!loggedInUser.hasConsentedToTerms) {
+      // If a user signed up in Auth0 and did not complete the New Account wizard
```

Comment:
Could you add a comment saying this check applies to all routes that use the component?

Reply:
Added in https://github.com/opentripplanner/otp-react-redux/pull/316/commits/032070e4e2ad1933a64dda4690310a539fb64cdf